src/models/user.model.js

```diff
@@ -171,7 +171,7 @@ userSchema.statics = {
 
     const users = results.map((result) => result.transform())
     const total = await User.find(find).count()
-    const pages = Math.ceil(total / perPage)
+    const pages = Math.ceil(total / Math.abs(perPage))
```

Comment:
This code executes each time when we fetch users, But you can see when we using `perPage = -1` we also changes the query too. So if you can do this operation in that if block, it would enhance the perfomance

Also this issue remains in other models too. Please take a look